File: nat/lvs_packet/nat-monitor/script/lib/report.py
import os
import sys
import urllib
import socket

gsms="hadoop_dxt_logsget"
gemail="hadoop_dxt_logsget_emailonly"
###################################################
def doAlarm(title,content,dosms=False):
	hostname = "["+socket.gethostname()+"] "
	baseurl = "http://alarms.ops.qihoo.net:8360/intfs/alarm_intf"
	if dosms:
		params = urllib.urlencode({'group_name':gsms, 'subject': hostname+title, 'content': hostname+content})
		urllib.urlopen(baseurl,params).read()
	else:  
		params = urllib.urlencode({'group_name':gemail, 'subject': hostname+title, 'content': hostname+content})
		urllib.urlopen(baseurl,params).read()

if __name__ == '__main__':
	title="Wangfeng test"
	content="Wangfeng test"
	doAlarm(title,content,False)

File: nat/lvs_packet/nat-monitor/script/lib/report.pl
#!/usr/bin/perl

use LWP;

$gsms="hadoop_dxt_logsget";
$gemail="hadoop_dxt_logsget_emailonly";

sub doAlarm {
        my $ua = LWP::UserAgent->new;
	my $hostname = `hostname`;
        my $title   = "[".$hostname."] ".$_[0];
        my $content = "[".$hostname."] ".$_[1];
	my $dosms = $_[2];
	
	$baseurl = "http://alarms.ops.qihoo.net:8360/intfs/alarm_intf";
	if ($dosms == 1){
                my $url = $baseurl."?group_name=$gsms&&subject=$title&content=$content";
                $response = $ua->get($url);
        }else{
		my $url = $baseurl."?group_name=$gemail&&subject=$title&content=$content";
		$response = $ua->get($url);
	}
}

